File: Problems 0100-0199/121. Best Time to Buy and Sell Stock/solution.cpp
class Solution {
    public:
        int maxProfit(vector<int>& prices) {
            int n = prices.size();
            if (n == 0) return 0;
    
            int minPrice = prices[0];
            int maxProfit = 0;
            for (int i = 1; i < n; i++) {
                int currProfit = prices[i]-minPrice;
                if (currProfit > maxProfit) {
                    maxProfit = currProfit;
                }
    
                if (prices[i] < minPrice) {
                    minPrice = prices[i];
                }
            }
    
            return maxProfit;
        }
    };

File: Problems 0100-0199/121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        if n == 0: return 0

        minPrice = prices[0]
        maxProfit = 0

        for i in range(1, n):
            if prices[i]-minPrice > maxProfit:
                maxProfit = prices[i]-minPrice
            
            minPrice = min(minPrice, prices[i])
        
        return maxProfit